src/legend.ts

```diff
@@ -56,14 +56,16 @@ const formatNumber = (d) => {
 }
 
 function rangeStep(domain: [number, number], index: number, bins: number = 9) {
-  if (index === 0) {
-    return domain[0];
-  } else if (index + 1 === bins) {
-    return domain[1];
-  } else {
-    const increment = (domain[1] - domain[0]) / bins;
-    return domain[0] + increment * index;
-  }
+    if(domain && Array.isArray(domain)) {
```

Comment:
Shouldn't we return something on an else case here? Will it break anything if rangeStep doesn't return anything?

Reply:
I am not able to reproduce a non array cases, but the error we had was on null, so I will add else case and return `null` to match up.